superset-frontend/src/dashboard/components/Header/index.jsx

```diff
@@ -52,8 +58,7 @@ const propTypes = {
   addSuccessToast: PropTypes.func.isRequired,
   addDangerToast: PropTypes.func.isRequired,
   addWarningToast: PropTypes.func.isRequired,
-  userId: PropTypes.number,
-  userEmail: PropTypes.string,
+  user: PropTypes.object,
```

Comment:
has a user type been defined yet?

Reply:
Not yet, I can work on one, especially since we only need id, email, and roles.

Reply:
Actually, this isn't tsx, so we can hold off.

Reply:
There is one already made, so I used it here: https://github.com/apache/superset/pull/15780/files#diff-28dce1dca6d6af7337727e88763b10a356e37ac5a15bd731d05077e89e0a8298R43